### PR TITLE
Performance tuning

### PR DESCRIPTION
**Pull request by mwiewior**
setting datafusion.execution.coalesce_batches to false

### CODE REVIEW

File: sequila/sequila-core/src/physical_planner/joins/interval_join.rs [hunk 1]
```diff
@@ -1046,20 +1044,25 @@ impl IntervalJoinStream {
...
             build_side
                 .hash_map
                 .get(*hash_val, start.value(i), end.value(i), |pos| {
-                    left_builder.append(pos as u32);
-                    right_builder.append(i as u32);
-                })
+                    pos_vect.push(pos as u32);
+                    i_vect.push(i as u32);
+                });
+            builder_left.append_slice(&pos_vect);
```
Comment (pkhamutou) [inline sequila/sequila-core/src/physical_planner/joins/interval_join.rs:1059]: so it's faster to push to a vector first and then to arrow array?

File: sequila/sequila-core/src/physical_planner/joins/interval_join.rs [hunk 2]
```diff
@@ -1046,20 +1044,28 @@ impl IntervalJoinStream {
...
+            rle_right.push(pos_vect.len() as u32);
+            builder_left.append_slice(&pos_vect);
+            pos_vect.clear();
         }
-
-        let left_indexes: UInt32Array = PrimitiveArray::new(left_builder.finish().into(), None);
-        let right_indexes: UInt32Array = PrimitiveArray::new(right_builder.finish().into(), None);
+        let left_indexes = builder_left.finish();
+        let mut index_right = Vec::with_capacity(left_indexes.len());
+        for i in 0..rle_right.len() {
```
Comment (pkhamutou) [inline sequila/sequila-core/src/physical_planner/joins/interval_join.rs:1063]: could you explain a bit how it works?

File: sequila/sequila-core/src/physical_planner/joins/interval_join.rs [hunk 3]
```diff
@@ -1046,20 +1044,28 @@ impl IntervalJoinStream {
...
+        let mut rle_right: Vec<u32> = Vec::with_capacity(self.hashes_buffer.len());
+        let mut pos_vect: Vec<u32> = Vec::with_capacity(100);
         for (i, hash_val) in self.hashes_buffer.iter().enumerate() {
             build_side
                 .hash_map
                 .get(*hash_val, start.value(i), end.value(i), |pos| {
-                    left_builder.append(pos as u32);
-                    right_builder.append(i as u32);
-                })
+                    pos_vect.push(pos as u32);
```
Comment (pkhamutou) [inline sequila/sequila-core/src/physical_planner/joins/interval_join.rs:1055]: why do we update only 1 position vector?

File: sequila/sequila-core/src/physical_planner/joins/interval_join.rs [hunk 4]
```diff
@@ -1046,20 +1044,28 @@ impl IntervalJoinStream {
         let start = evaluate_as_i32(self.right_interval.start(), &state.batch)?;
         let end = evaluate_as_i32(self.right_interval.end(), &state.batch)?;
 
-        let mut left_builder = UInt32BufferBuilder::new(0);
-        let mut right_builder = UInt32BufferBuilder::new(0);
+        let mut builder_left = PrimitiveBuilder::<UInt32Type>::new();
 
+        let mut rle_right: Vec<u32> = Vec::with_capacity(self.hashes_buffer.len());
```
Comment (pkhamutou) [inline sequila/sequila-core/src/physical_planner/joins/interval_join.rs:1049]: I have come up with this solution, which is better then what we have and it seems like it gives slightly better performance then yours:
```rust
        // pre-allocate 2 vectors of at least the batch size
        let mut left_indexes: Vec<u32> = Vec::with_capacity(self.hashes_buffer.len());
        let mut right_indexes: Vec<u32> = Vec::with_capacity(self.hashes_buffer.len());

        for (i, hash_val) in self.hashes_buffer.iter().enumerate() {
            build_side
                .hash_map
                .get(*hash_val, start.value(i), end.value(i), |pos| {
                    left_indexes.push(pos as u32);
                    right_indexes.push(i as u32);
                })
        }

        let left_indexes: UInt32Array = left_indexes.into(); 
        let right_indexes: UInt32Array = right_indexes.into();
```

Comment (mwiewior) [reply sequila/sequila-core/src/physical_planner/joins/interval_join.rs:1049]: Let's benchmark it!

**Comment by mwiewior** [comment]
Resolves #41

**Pull request merged**